pkg/provider/provider.go

```diff
@@ -600,8 +599,16 @@ func ConfigureProvider(s *schema.ResourceData) (interface{}, error) {
 		config.Region = v.(string)
 	}
 
-	if v, ok := s.GetOk("validate_default_parameters"); ok && v.(bool) {
-		config.ValidateDefaultParameters = gosnowflake.ConfigBoolTrue
+	if v := s.Get("validate_default_parameters").(string); v != resources.BooleanDefault {
+		parsed, err := resources.BooleanStringToBool(v)
```

Comment:
IMO we shouldn't depend on the resources package here. Please extract the special values and than we can reuse them in both packages (and datasources too if needed).

Reply:
Done.